common/persistence/tasks.go

```diff
@@ -320,116 +338,556 @@ func (d *DecisionTask) GetVersion() int64 {
+	}
+	switch t.TaskType {
+	case TransferTaskTypeActivityTask:
+		return &ActivityTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+			TargetDomainID:     t.TargetDomainID,
+			TaskList:           t.TaskList,
+			ScheduleID:         t.ScheduleID,
+		}, nil
+	case TransferTaskTypeDecisionTask:
+		return &DecisionTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+			TargetDomainID:     t.TargetDomainID,
+			TaskList:           t.TaskList,
+			ScheduleID:         t.ScheduleID,
+		}, nil
+	case TransferTaskTypeCloseExecution:
+		return &CloseExecutionTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+		}, nil
+	case TransferTaskTypeRecordWorkflowStarted:
+		return &RecordWorkflowStartedTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+		}, nil
+	case TransferTaskTypeResetWorkflow:
+		return &ResetWorkflowTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+		}, nil
+	case TransferTaskTypeRecordWorkflowClosed:
+		return &RecordWorkflowClosedTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+		}, nil
+	case TransferTaskTypeRecordChildExecutionCompleted:
+		return &RecordChildExecutionCompletedTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+			TargetDomainID:     t.TargetDomainID,
+			TargetWorkflowID:   t.TargetWorkflowID,
+			TargetRunID:        t.TargetRunID,
+		}, nil
+	case TransferTaskTypeUpsertWorkflowSearchAttributes:
+		return &UpsertWorkflowSearchAttributesTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+		}, nil
+	case TransferTaskTypeStartChildExecution:
+		return &StartChildExecutionTask{
+			WorkflowIdentifier: workflowIdentifier,
+			TaskData:           taskData,
+			TargetDomainID:     t.TargetDomainID,
+			TargetWorkflowID:   t.TargetWorkflowID,
+			InitiatedID:        t.ScheduleID,
+		}, nil
+	case TransferTaskTypeCancelExecution:
+		return &CancelExecutionTask{
+			WorkflowIdentifier:      workflowIdentifier,
+			TaskData:                taskData,
+			TargetDomainID:          t.TargetDomainID,
+			TargetWorkflowID:        t.TargetWorkflowID,
+			TargetRunID:             t.TargetRunID,
+			InitiatedID:             t.ScheduleID,
+			TargetChildWorkflowOnly: t.TargetChildWorkflowOnly,
+		}, nil
+	case TransferTaskTypeSignalExecution:
+		return &SignalExecutionTask{
+			WorkflowIdentifier:      workflowIdentifier,
+			TaskData:                taskData,
+			TargetDomainID:          t.TargetDomainID,
+			TargetWorkflowID:        t.TargetWorkflowID,
+			TargetRunID:             t.TargetRunID,
+			InitiatedID:             t.ScheduleID,
+			TargetChildWorkflowOnly: t.TargetChildWorkflowOnly,
+		}, nil
+	default:
+		return nil, fmt.Errorf("unknown task type: %d", t.TaskType)
+	}
+}
+
+func ToTransferTaskInfo(task Task) (*TransferTaskInfo, error) {
```

Comment:
Instead of a helper function with switch block, could we add `ToTransferTaskInfo()` to `Task` interface? That would make it future proof. Mappers have proven to be problematic. At least we can eliminate one problem which is someone forgetting to update this file when new types/fields introduced. Interface compliance would break the build